cpp/models/lct_secir/parameters_io.h

```diff
@@ -0,0 +1,317 @@
+    ScalarType timeInfectedCritical   = model.parameters.template get<TimeInfectedCritical>();
+    ScalarType scale_asymptomatic     = 1 / (1 - model.parameters.template get<RecoveredPerInfectedNoSymptoms>());
+
+    ScalarType min_offset_needed = std::floor(-timeInfectedSymptoms - timeInfectedSevere - timeInfectedCritical);
+    ScalarType max_offset_needed = std::ceil(timeExposed + timeInfectedNoSymptoms);
+
+    bool min_offset_needed_avail = false;
+    bool max_offset_needed_avail = false;
+
+    // Go through the entries of rki_data and check if date is needed for calculation. Confirmed cases are scaled by scale_confirmed_cases.
+    for (auto&& entry : rki_data) {
+        int offset = get_offset_in_days(entry.date, date);
+        if ((offset >= min_offset_needed) && (offset <= max_offset_needed)) {
+            // Add confirmed cases at date to compartment Recovered.
+            if (offset == 0) {
+                init[LctState::template get_first_index<InfectionState::Recovered>()] +=
+                    scale_confirmed_cases * entry.num_confirmed;
+            }
+
+            // Compute initial values for compartment InfectedNoSymptoms.
+            if (offset >= 0 && offset <= std::ceil(timeInfectedNoSymptoms)) {
+                ScalarType TCi = timeInfectedNoSymptoms /
+                                 LctState::template get_num_subcompartments<InfectionState::InfectedNoSymptoms>();
+                int idx_Cn = LctState::template get_first_index<InfectionState::InfectedNoSymptoms>() +
+                             LctState::template get_num_subcompartments<InfectionState::InfectedNoSymptoms>() - 1;
+                for (int i = 0;
+                     i < (int)LctState::template get_num_subcompartments<InfectionState::InfectedNoSymptoms>(); i++) {
+                    if (offset == std::floor(i * TCi)) {
+                        init[idx_Cn - i] -= (1 - (i * TCi - std::floor(i * TCi))) * scale_asymptomatic *
+                                            scale_confirmed_cases * entry.num_confirmed;
+                    }
+                    if (offset == std::ceil(i * TCi)) {
+                        init[idx_Cn - i] -= (i * TCi - std::floor(i * TCi)) * scale_asymptomatic *
+                                            scale_confirmed_cases * entry.num_confirmed;
+                    }
+                    if (offset == std::floor((i + 1) * TCi)) {
+                        init[idx_Cn - i] += (1 - ((i + 1) * TCi - std::floor((i + 1) * TCi))) * scale_asymptomatic *
+                                            scale_confirmed_cases * entry.num_confirmed;
+                    }
+                    if (offset == std::ceil((i + 1) * TCi)) {
+                        init[idx_Cn - i] += ((i + 1) * TCi - std::floor((i + 1) * TCi)) * scale_asymptomatic *
+                                            scale_confirmed_cases * entry.num_confirmed;
+                    }
```

Comment:
Why do you do the scaling in every if statement separately? I would find it clearer if you would do the interpolation in one step and after that scale accordingly. But I'm actually not sure if there is a nice way to do that or if this makes it just more complicated. What do you think?

Reply:
Alternatively, I could iterate through the entire timeseries again and multiply by scale_confirmed_cases. But I don't think that's better. Maybe there is a more elegant solution for this? I also thought that this make the meaning of the parameter scale_confirmed_cases clear because it is used combined with num_confirmed.